sdk/core/core-lro/src/poller.ts

```diff
@@ -0,0 +1,133 @@
+import { EventEmitter } from "events";
+import { delay, HttpOperationResponse, RequestOptionsBase, ServiceClient } from "@azure/core-http";
+
+export interface PollerOptionalParameters<States> {
+  manual?: boolean;
+  intervalInMs?: number;
+  state?: States;
```

Comment:
should there be an explicit poller state as well? e.g. Unstarted, Started, Done, Error or something?

Reply:
That's a good question. But, at that point, I'd rather have an explicit state defined by us, but not the one that core-arm has (the one with the two spellings), rather one that can include "pollError", so that we could just pass this state to this.emit. What do you think?

Reply:
I don't need to align to core-arm, but I do think it should be clear what part of its lifecycle a poller is in. Assuming we can't restart a poller once it's been stopped, then it should be a pretty simple state machine. 

I think we could derive some of this today, since we have `stopped` and presumably previousResponse would allow for checking `isDone()`, but it's not clear that terminal error states would be captured or if `initialResponse` is sufficient to determine started.

For example, what if the initial request was sent and then the poller was serialized before the initial response had been received. It might look like it was never started at all and duplicate the operation when rehydrated.

So we at least need to track `started` and prevent sending the initial request twice, I think.

Reply:
What do you think of unifying both states? If so, how would that look? Would something like `started, finished, aborted, cancelled, stopped` ?

Reply:
yeah I think we could make it an atomic value like that (using a string literal union or an enum)

One question,  is aborted semantically different than cancelled?